src/trainer.py

```diff
@@ -350,9 +355,13 @@ def _setup_training(
             task_info["last_log"] = time.time()
 
         # Metric bookkeeping
-        all_metrics = [task.val_metric for task in tasks] + ["micro_avg", "macro_avg"]
```

Comment:
This change seems like it could break quite a lot. Split it off if you can, otherwise run as many tests as you can, and read through the full trainer file for possible issues.